app/components/layout/sidebar/index.tsx

```diff
@@ -34,45 +56,135 @@ export default function Sidebar({ user }: Props) {
     },
   ];
 
+  const collapseSidebar = useCallback(() => {
+    setIsSidebarCollapsed(true);
+  }, [setIsSidebarCollapsed]);
+  const uncollapseSidebar = useCallback(() => {
+    setIsSidebarCollapsed(false);
+  }, [setIsSidebarCollapsed]);
+
+  useEffect(() => {
+    setHasMounted(true);
+    if (hasMounted) {
+      if (localStorage.getItem("sidebarSwitchIsClicked")) {
+        setSidebarSwitchIsClicked(
+          localStorage.getItem("sidebarSwitchIsClicked") === "true"
+        );
+      }
+      if (localStorage.getItem("isSidebarCollapsed")) {
+        setIsSidebarCollapsed(
+          localStorage.getItem("isSidebarCollapsed") === "true"
+        );
+      }
+      const sidebar = document.getElementById("main-navigation") as HTMLElement;
+      if (sidebarSwitchIsClicked === false) {
+        sidebar.addEventListener("mouseenter", uncollapseSidebar);
+        sidebar.addEventListener("mouseleave", collapseSidebar);
+      }
+    }
+  });
+
+  function maintainUncollapsedState() {
+    const sidebar = document.getElementById("main-navigation") as HTMLElement;
+    if (sidebarSwitchIsClicked === false) {
+      sidebar.removeEventListener("mouseenter", uncollapseSidebar);
+      sidebar.removeEventListener("mouseleave", collapseSidebar);
+    } else {
+      sidebar.addEventListener("mouseenter", uncollapseSidebar);
+      sidebar.addEventListener("mouseleave", collapseSidebar);
+    }
+    setSidebarSwitchIsClicked(!sidebarSwitchIsClicked);
+  }
+
+  if (!hasMounted) {
+    return null;
+  }
+
   return (
-    <div className=" flex h-full flex-col ">
-      <div>
-        <Link to="." title="Home">
+    <aside
+      id="main-navigation"
+      className={tw(
+        `sticky top-0 flex h-screen w-80 flex-col border-r border-gray-200 px-6 py-8 transition-all duration-200 ease-linear`,
+        isSidebarCollapsed ? "collapsed" : ""
+      )}
+    >
+      <div className="navigation-header flex items-center justify-between">
+        <Link to="." title="Home" className="flex items-center">
           <img
-            src="/images/logo-full-color(x2).png"
+            src="/images/shelf-symbol.png"
             alt="Shelf Logo"
-            className="h-[30px]"
+            className="mx-1.5 inline h-[32px]"
           />
+          <span className="logo-text transition duration-200 ease-linear">
+            <svg
```

Comment:
Lets move this to a separate component as those big svg's make the component look very messy.

Reply:
Should I move this to /components/icons/library or create a separate component in /components/icons

Reply:
Move the SVG to the library and just render it within the span in the view. That should be a good enough solution.